spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+RSpec.describe 'post index page', type: :feature do
+  describe 'Post Index' do
+    before(:each) do
+      @user1 = User.create(name: 'Buntu', photo: 'https://unsplash.com/photos/WNoLnJo7tS8', bio: 'Lecturer', posts_counter: 1)
+
+      visit root_path
+
+      @post1 = Post.create(title: 'First post', text: 'This is my first post', comments_counter: 1, likes_counter: 1, author_id: @user1.id)
+      @post2 = Post.create(title: 'Second post', text: 'This is my second post', comments_counter: 1, likes_counter: 1, author_id: @user1.id)
+      @post3 = Post.create(title: 'Third post', text: 'This is my third post', comments_counter: 1, likes_counter: 1, author_id: @user1.id)
+      @comment1 = Comment.create(text: 'First comment', author_id: @user1.id, post_id: @post1.id)
+      @comment2 = Comment.create(text: 'Second comment', author_id: @user1.id, post_id: @post1.id)
+      @comment3 = Comment.create(text: 'Third comment', author_id: @user1.id, post_id: @post1.id)
+    
+      visit user_posts_path(@user1, @post1)
```

Comment:
- [ ] For this test you are visiting the post show page, instead of the post index page `(user_posts_path(@user1))`. Kindly use the correct route for these tests